projects/ngx-d3timeline/src/lib/scale.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable } from '@angular/core';
+import { State } from './store/state';
+import { TimeScale, BandScale } from './scale-types';
+import { Orientation } from './orientation';
+import { scaleBand, scaleTime } from 'd3-scale';
+import { TimelineEvent } from './timeline-event';
+import { min, max } from 'd3-array';
+import { TimelineView } from './view/timeline-view';
+
+@Injectable({ providedIn: 'root' })
+export class ScaleService {
+  rescaleTime(state: State, event: any): TimeScale {
+    const timeScale = this.configureTimeScale(state);
```

Comment:
I recall you doing this last night, but not clear to me why state.timeScale is not already up to date - why does it need recalcalcing here I assume with the same previous state values? Does it imply the state was stale?

Reply:
Its because we need to rescale the scaleTime that was generated from the data,view, and orientation in the state. The current scaleTime in the store already has the rescale applied to it. This is only happening because we've put the scales in the state.

Reply:
I see, maybe call this local variable `unscaledTimeScale` to differentiate it from the timeScale in the state?

Bit of a tongue twister but it is meaningful.